### ISSUE TITLE
routes helper ignores mount point when testing

### ISSUE THREAD

**Issue by mereghost** [hanami/hanami#571]
Apparently the routes helper ignores the mount point of the application when running tests.

The following code should be enough to reproduce the error:

apps/web/controllers/stuffs/index.rb
```ruby
module Web::Controllers::Stuffs
  class Index
    include Web::Action
    def call(params)
      self.headers.merge!({'Location' => String.new(routes.stuff_url(1)))
      self.body = 'Irrelevant'
    end
  end
end
```
apps/web/config/routes.rb
```ruby
resources :stuffs
```

config/environment.rb
```ruby
# ...
Hanami::Container.configure { mount Web::Application, at: '/my/path' }
```

spec/web/features/return_a_weird_location_header_spec.rb
```ruby
require 'features_helper'

describe 'return a weird location header' do
  it 'respects the mount path' do
    visit '/my/path/stuffs'
    expect(page.driver.response.headers['Location']).to match /my\/path/
  end
end
```

I have tested it with Capybara (and later with Rack::Test only) both on stable and head. It seems that, while the router recognizes the mount point, as the request is successful, it does not pass this knowledge down to the controller.

Let me know if any further info is necessary.

**Comment by Erol** [comment]
Hi @jodosha, would you mind if I take a stab at this? I was able to trace why this is happening and can open a WIP PR where we can discuss how it can be resolved.

**Comment by jodosha** [comment]
@mereghost Hi Marcello, can you please check if #579 fixes your problem? Thanks!

**Comment by mereghost** [comment]
@jodosha Hi Luca, it works. Thanks for taking a stab at it @Erol and @jodosha  =)

**Comment by jodosha** [comment]
Fixed by #579
